### PR TITLE
Fix recover (add more signature validation)

### PR DESCRIPTION
**Pull request by fanatid**
See #63

### CODE REVIEW

**Comment by fanatid** [comment]
Tests fails, I'll fix in few hours.

File: lib/elliptic/verify.js [hunk 1]
```diff
@@ -30,5 +32,9 @@ exports.verifySync = function (msg, signature, publicKey) {
   }
 
   var sigObj = {r: signature.slice(0, 32), s: signature.slice(32, 64)}
+  if (new BN(sigObj.s).cmp(ec.nh) === 1 || new BN(sigObj.r).isZero() || new BN(sigObj.s).isZero()) {
```
Comment (wanderer) [inline lib/elliptic/verify.js]: Should this or the equivalent go directly in elliptic?

File: lib/elliptic/recover.js [hunk 2]
```diff
@@ -27,6 +29,10 @@ exports.recoverSync = function (msg, signature, recovery) {
 
   try {
     var sigObj = {r: signature.slice(0, 32), s: signature.slice(32, 64)}
+    if (new BN(sigObj.r).isZero() || new BN(sigObj.s).isZero()) {
```
Comment (wanderer) [inline lib/elliptic/recover.js:32]: would it not be better just todo `verify` and move this to verify?

Comment (fanatid) [inline lib/elliptic/recover.js:32]: No, different error messages here :(

File: lib/elliptic/verify.js (hunk 1 above)
Comment (fanatid) [inline lib/elliptic/verify.js]: Check elliptic and `ec.verify` already has `isZero` checks.

**Comment by wanderer** [comment]
lgtm

**Pull request merged**